backend/entityservice/views/project.py

```diff
@@ -347,39 +350,44 @@ def upload_clk_data_binary(project_id, dp_id, raw_stream, count, size=128):
     return receipt_token
 
 
-def upload_json_clk_data(dp_id, clk_json, parent_span):
+def upload_json_clk_data(dp_id, clk_json, parent_span, uses_blocking):
     """
     Convert user provided encodings from json array of base64 data into
     a newline separated file of base64 data.
 
     Note this implementation is non-streaming.
     """
-    if 'clks' not in clk_json or len(clk_json['clks']) < 1:
-        safe_fail_request(400, message="Missing CLKs information")
+    abort_if_inconsistent_upload(uses_blocking, clk_json)
+
+    # now we need to know element name - clks or clknblocks
+    is_valid_clks = not uses_blocking and 'clks' in clk_json
+    element = 'clks' if is_valid_clks else 'clknblocks'
 
     receipt_token = generate_code()
 
     filename = Config.RAW_FILENAME_FMT.format(receipt_token)
-    logger.info("Storing user {} supplied clks from json".format(dp_id))
+    logger.info("Storing user {} supplied {} from json".format(dp_id, element))
 
     with opentracing.tracer.start_span('splitting-json-clks', child_of=parent_span) as span:
-        count = len(clk_json['clks'])
-        span.set_tag("clks", count)
-        data = b''.join(''.join(clk.split('\n')).encode() + b'\n' for clk in clk_json['clks'])
+        count = len(clk_json[element])
+        span.set_tag(element, count)
+
+    logger.info(f"Received {count} encodings.")
```

Comment:
If the element type was `clksnblocks` will this statement be correct?

Reply:
yes